File: app/src/main/java/com/vdovin/spacex/screen/main/structure/LaunchesView.java
package com.vdovin.spacex.screen.main.structure;

import com.vdovin.spacex.base.BaseView;
import com.vdovin.spacex.database.model.SpaceX;

import java.util.List;

import io.reactivex.Observable;

public interface LaunchesView extends BaseView {

    void displayLaunches(List<SpaceX> launches);

    void openDetails(SpaceX spaceX);

    Observable<SpaceX> launchClicked();
}

File: app/src/main/java/com/vdovin/spacex/screen/main/LaunchesView.kt
package com.vdovin.spacex.screen.main

import com.vdovin.spacex.base.BaseView
import com.vdovin.spacex.database.model.SpaceX




interface LaunchesView : BaseView {

    fun displayLaunches(launches: List<SpaceX>)

    fun openDetails(spaceX: SpaceX)

    fun launchClicked(launchClickListener : (SpaceX) -> Unit)

}